### PR TITLE
[WIP] Doc Only: Update gitrepository.md about Github Apps

### PR DESCRIPTION
**Pull request by jeffkala**
# Closes: N/A
# What's Changed

Be more explict that the currently functionalities within Git Repos don't support GitHub apps.

# TODO

- [x] Explanation of Change(s)
- [x] Documentation Updates (when adding/changing features)

### CODE REVIEW

**Comment by gsnider2195** [comment]
By github apps do you mean something like https://github.com/marketplace/gitguardian

**Comment by jeffkala** [comment]
More like custom apps to try and solve not needing a service account.
https://docs.github.com/en/apps/creating-github-apps/creating-github-apps/about-apps

**Comment by gsnider2195** [comment]
> More like custom apps to try and solve not needing a service account. https://docs.github.com/en/apps/creating-github-apps/creating-github-apps/about-apps

I see. Maybe it would be clearer to just say something like nautobot only supports repositories that can be cloned with the `git` cli app using `git clone`?

**Comment by bryanculver** [comment]
I'm confused here. Along with #3151, where is the use case where your Git Repository is a Github App?

I can see the case for documentation stating we don't support Github App authentication but in review #3151 I'm not seeing how the REST API token can be used to perform a `git clone` onto the filesystem.

**Comment by bryanculver** [comment]
Marking as WIP until feedback is addressed.

**Comment by jeffkala** [comment]
This is all about the management and accessibility to repositories.  GH Apps provides some safeguards for this. Ultimately the goal of this PR is not to provide a fix or recommendation beyond just mentioning it in the docs that we don't support the functionality.  I'm happy to add additional details if needed.

**Comment by jeffkala** [comment]
> > More like custom apps to try and solve not needing a service account. https://docs.github.com/en/apps/creating-github-apps/creating-github-apps/about-apps
> 
> I see. Maybe it would be clearer to just say something like nautobot only supports repositories that can be cloned with the `git` cli app using `git clone`?

I'm open to changing the terminology but do think GH Apps should be directly called out.

File: nautobot/docs/models/extras/gitrepository.md [hunk 1]
```diff
@@ -14,6 +14,9 @@ Some text-based content is more conveniently stored in a separate Git repository
 
 When defining a Git repository for Nautobot to consume, the `name`, `remote URL`, and `branch` parameters are mandatory - the name acts as a unique identifier, and the remote URL and branch are needed for Nautobot to be able to locate and access the specified repository. Additionally, if the repository is private you may specify a `token` and any associated `username` that can be used to grant access to the repository.
 
+!!! note
+    GitHub Apps are not currently supported. GitHub Apps are not specifically focused on the workflow of creating/updating/deleting files. They can indeed perform these actions, but in order to accomplish that the client side MUST hit GitHub APIs instead of using standard command line or Python Git library tooling. At this time this functionality does not exist. 
+
```
Comment (jeffkala) [inline nautobot/docs/models/extras/gitrepository.md:19]:
```suggestion
!!! note
    GitHub Apps are not currently supported. GitHub Apps are not specifically focused on the workflow of creating/updating/deleting files.  Nautobot only supports repositories that can be cloned using standard git command line using `git clone`. 

```

Comment (bryanculver) [inline nautobot/docs/models/extras/gitrepository.md:19]:
```suggestion
!!! note
    Nautobot currently only supports repositories that can be cloned using the standard git command line, `git clone`. This means App-style integrations like GitHub Apps are not currently supported, as their workflow of managing files leverages a REST API.
```


@jeffkala?

Comment (jeffkala) [reply nautobot/docs/models/extras/gitrepository.md]: committed suggestion. Thanks!

**Pull request merged**